controller_manager/src/controller_manager.cpp

```diff
@@ -62,33 +63,66 @@ ControllerManager::ControllerManager(
   loader_(std::make_shared<pluginlib::ClassLoader<controller_interface::ControllerInterface>>(
       kControllerInterfaceName, kControllerInterface))
 {
+  realtime_callback_group_ = create_callback_group(
+    rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
+  services_callback_group_ = create_callback_group(
+    rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);
```

Comment:
I like this two callback group idea, but `services_callback_group_` might not be a good name, because we could simplify some callbacks if we made them RT friendly and put them in the `realtime_callback_group_`.

The whole stop/start controller is unnecessarily complex because we receive these callbacks when `update` could be running.

Reply:
The main reason for the groups is that if they do not exist than the services and update are blocking each other, i.e., the multi-threading is not existing.

Nevertheless, I can change the name if you have another proposal.

Reply:
I have alternatives but none that are great.. non_realtime or auxiliar.  Main and secondary, but all of them require a nice comment block explaining what are they.

Reply:
Let's go with real-time and non-realtime.

Reply:
What about *best effort* and *deterministic* (or *real-time safe*)?

Reply:
I like deterministic, but best effort might be confused with the Reliability QoS setting of ROS2.
Regardless, a few comments should clear any confusion.

Reply:
> best effort might be confused with the Reliability QoS setting of ROS2.

that would be avoided with a one-line comment, no?

In RT nomenclature, *best effort* just happens to be the name if something isn't actually deterministic. But you already know that I assume.

Reply:
Actually I wasn't aware, it makes more sense then.And again a short comment might clear any doubts.

Reply:
OK i renamed the variables to: `deterministic_callback_group_` and `best_effort_callback_group_`